File: src/main/java/com/app/modules/Calculator.java
package com.app.modules;

import java.math.BigDecimal;
import java.math.RoundingMode;

/**
 * Calculator module for arithmetic operations
 * Migrated from COBOL CALCULATOR module
 */
public class Calculator {
    
    // Constants migrated from calculator.cpy
    public static final BigDecimal MAX_VALUE = new BigDecimal("999999.99");
    public static final BigDecimal MIN_VALUE = new BigDecimal("-999999.99");
    
    /**
     * Adds two decimal numbers with precision
     * This method replicates the functionality of the COBOL CALCULATOR program
     * 
     * @param num1 First number
     * @param num2 Second number
     * @return The sum of the two numbers, constrained to MIN_VALUE and MAX_VALUE
     */
    public static BigDecimal add(BigDecimal num1, BigDecimal num2) {
        if (num1 == null) num1 = BigDecimal.ZERO;
        if (num2 == null) num2 = BigDecimal.ZERO;
        
        BigDecimal result = num1.add(num2).setScale(2, RoundingMode.HALF_UP);
        
        // Apply constraints from COBOL constants
        if (result.compareTo(MAX_VALUE) > 0) {
            return MAX_VALUE;
        } else if (result.compareTo(MIN_VALUE) < 0) {
            return MIN_VALUE;
        }
        
        return result;
    }
}

File: CobolAppNet/src/CobolAppNet/Modules/Calculator.cs
namespace CobolApp.Net.Modules
{
    /// <summary>
    /// Calculator module for arithmetic operations
    /// Migrated from COBOL CALCULATOR module
    /// </summary>
    public static class Calculator
    {
        // Constants migrated from calculator.cpy
        public static readonly decimal MaxValue = 999999.99m;
        public static readonly decimal MinValue = -999999.99m;

        /// <summary>
        /// Adds two decimal numbers with precision
        /// This method replicates the functionality of the COBOL CALCULATOR program
        /// </summary>
        /// <param name="num1">First number</param>
        /// <param name="num2">Second number</param>
        /// <returns>The sum of the two numbers, constrained to MinValue and MaxValue</returns>
        public static decimal Add(decimal num1, decimal num2)
        {
            decimal result = num1 + num2;

            // Apply constraints from COBOL constants
            if (result > MaxValue)
            {
                return MaxValue;
            }
            else if (result < MinValue)
            {
                return MinValue;
            }

            return Math.Round(result, 2);
        }
    }
}
